### PR TITLE
#gh11334: show lastName/firstName from AD_User on Search Assistant.

### PR DESCRIPTION
**Pull request by adi-stefan**
https://github.com/metasfresh/metasfresh/issues/11334

### CODE REVIEW

File: backend/de.metas.acct.base/src/main/sql/postgresql/ddl/views/c_bpartner_adv_search_v.sql [hunk 1]
```diff
@@ -1,23 +1,24 @@
-DROP VIEW IF EXISTS C_BPartner_Adv_Search_v;
+DROP VIEW IF EXISTS C_BPartner_Adv_Search_v
+;
+
 CREATE OR REPLACE VIEW C_BPartner_Adv_Search_v AS
 SELECT p.c_bpartner_id,
        pl.c_bpartner_location_id,
+       COALESCE(u.ad_user_id,-1) as userID,
```
Comment (teosarca) [inline backend/de.metas.acct.base/src/main/sql/postgresql/ddl/views/c_bpartner_adv_search_v.sql:7]: use AD_User_ID

File: backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/descriptor/AdvancedSearchBPartnerProcessor.java [hunk 2]
```diff
@@ -24,25 +24,57 @@
...
+	private final SqlViewFactory sqlViewFactory;
+
+	public AdvancedSearchBPartnerProcessor(final SqlViewFactory sqlViewFactory)
+	{
+		this.sqlViewFactory = sqlViewFactory;
+	}
 
 	@Override
-	public void processSelection(final Document document, final String bpartnerFieldName, final String selectionIdStr)
+	public void processSelection(final WindowId windowId, final Document document, final String bpartnerFieldName, final String selectionIdStr)
```
Comment (teosarca) [inline backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/descriptor/AdvancedSearchBPartnerProcessor.java:52]: fine for me to introduce WindowId param,
but you could use document.getDocumentPath().getWindowId()

Comment (adi-stefan) [reply backend/de.metas.ui.web.base/src/main/java/de/metas/ui/web/window/descriptor/AdvancedSearchBPartnerProcessor.java:52]: Not really. windowId is the ID of the search assistant window, while document is the sales order document to be populated.
image

File: backend/de.metas.acct.base/src/main/sql/postgresql/ddl/views/c_bpartner_adv_search_v.sql (hunk 1 above)
Comment (adi-stefan) [reply backend/de.metas.acct.base/src/main/sql/postgresql/ddl/views/c_bpartner_adv_search_v.sql:7]: As discussed changed to `C_BP_Contact_ID`

**Pull request merged**